src/Sylius/Component/Resource/spec/Repository/InMemoryRepositorySpec.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Resource\Repository;
+
+use Pagerfanta\Pagerfanta;
+use PhpSpec\ObjectBehavior;
+use spec\Sylius\Component\Resource\Fixtures\SampleResourceInterface;
+use Sylius\Component\Resource\Exception\UnsupportedMethodException;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\Component\Resource\Repository\InMemoryRepository;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+require_once __DIR__ . '/../Fixtures/SampleResourceInterface.php';
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class InMemoryRepositorySpec extends ObjectBehavior
+{
+    function let(SampleResourceInterface $book, SampleResourceInterface $shirt)
+    {
+        $this->beConstructedWith(SampleResourceInterface::class);
+
+        $book->getId()->willReturn(10);
+        $book->getName()->willReturn('Book');
+        $book->getRating()->willReturn(5);
+
+        $shirt->getId()->willReturn(5);
+        $shirt->getName()->willReturn('Shirt');
+
+        $this->add($book);
+        $this->add($shirt);
```

Comment:
We shouldn't 'load' sample data in `let`. The examples should be complete, but when I look at https://github.com/Sylius/Sylius/pull/3617/files#diff-cba4107abbc619c9dfb57a6d36956923R154 it's confusing, because I don't know why `$book` is returned since it wasn't even added to the repository.